lib/api/include/irods/ticketAPI.hpp

```diff
@@ -327,79 +202,245 @@ namespace irods::administration::ticket
-    int admin_create_write_ticket(RxComm& conn, std::string_view obj_path);
-
-    int admin_remove_usage_restriction(RxComm& conn, std::string_view ticket_name);
-    int admin_remove_usage_restriction(RxComm& conn, int ticket_ID);
-
-    int admin_remove_write_file_restriction(RxComm& conn, std::string_view ticket_name);
-    int admin_remove_write_file_restriction(RxComm& conn, int ticket_ID);
-
-    int admin_remove_write_byte_restriction(RxComm& conn, std::string_view ticket_name);
-    int admin_remove_write_byte_restriction(RxComm& conn, int ticket_ID);
-
-    int admin_set_usage_restriction(RxComm& conn, std::string_view ticket_name, int numUses);
-    int admin_set_usage_restriction(RxComm& conn, int ticket_ID, int numUses);
-
-    int admin_set_write_file_restriction(RxComm& conn, std::string_view ticket_name, int numUses);
-    int admin_set_write_file_restriction(RxComm& conn, int ticket_ID, int numUses);
-
-    int admin_set_write_byte_restriction(RxComm& conn, std::string_view ticket_name, int numUses);
-    int admin_set_write_byte_restriction(RxComm& conn, int ticket_ID, int numUses);
-
-    int admin_add_user(RxComm& conn, std::string_view ticket_name, std::string_view user);
-    int admin_add_user(RxComm& conn, int ticket_ID, std::string_view user);
+    void delete_ticket(admin_tag, RxComm& conn, std::string_view ticket_name);
+    void delete_ticket(admin_tag, RxComm& conn, int ticket_ID);
 
-    int admin_remove_user(RxComm& conn, std::string_view ticket_name, std::string_view user);
-    int admin_remove_user(RxComm& conn, int ticket_ID, std::string_view user);
-
-    int admin_add_group(RxComm& conn, std::string_view ticket_name, std::string_view group);
-    int admin_add_group(RxComm& conn, int ticket_ID, std::string_view group);
-
-    int admin_remove_group(RxComm& conn, std::string_view ticket_name, std::string_view group);
-    int admin_remove_group(RxComm& conn, int ticket_ID, std::string_view group);
-
-    int admin_add_host(RxComm& conn, std::string_view ticket_name, std::string_view host);
-    int admin_add_host(RxComm& conn, int ticket_ID, std::string_view host);
-
-    int admin_remove_host(RxComm& conn, std::string_view ticket_name, std::string_view host);
-    int admin_remove_host(RxComm& conn, int ticket_ID, std::string_view host);
+    // These functions are the same as the above, but it does them as an admin of the system
 
-    int admin_delete_ticket(RxComm& conn, std::string_view ticket_name);
-    int admin_delete_ticket(RxComm& conn, int ticket_ID);
+    void add_ticket_constraint(RxComm& conn, constraint::constraint& constraints);
+    void remove_ticket_constraint(RxComm& conn, constraint::constraint& constraints);
+
+    void add_ticket_constraint(admin_tag, RxComm& conn, constraint::constraint& constraints);
+    void remove_ticket_constraint(admin_tag, RxComm& conn, constraint::constraint& constraints);
+
+    // OLD ADMIN PERMISSION CONTROL
```

Comment:
I'm guessing you're planning on deleting these lines once the implementation is closer to being finished?

Reply:
Yup! I did so in the commit I just pushed